http/middleware.go

```diff
@@ -89,6 +89,18 @@ func processRequest(tx types.Transaction, req *http.Request) (*types.Interruptio
 		}
 	}
 
+	// Calling tx.RuleEngineStatus() and tx.RequestBodyAccessible() is possible to
+	// anticipate some checks performed inside ProcessRequestBody(), avoiding
+	// to call the latter if no inspections are going to happen.
+	// It is performed here as a matter of example. It is recommended to avoid doing it
+	// if not strictly needed for server/proxy side actions.
+	switch {
+	case tx.RuleEngineStatus() == types.RuleEngineOff:
```

Comment:
I think if `tx.RuleEngineStatus() == types.RuleEngineOff` I would avoid the wrapping at all because not rules are evaluated and hence nothing different than the wrapped handler should be executed. Probably early termination in https://github.com/corazawaf/coraza/blob/v3/dev/http/middleware.go#L106 and not considering that case in here.

Reply:
Yeah - unless gating at a higher level, there doesn't seem to be a reason to add these branches since `tx.ProcessRequestBody` itself should be effectively a no-op already.